### PR TITLE
test: add DatastoreException assertions in integration tests

### PR DESCRIPTION
**Pull request by kolea2**

### CODE REVIEW

**Review by cindy-peng — APPROVED** [review]

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java [hunk 1]
```diff
@@ -1620,7 +1620,7 @@ public Integer run(DatastoreReaderWriter transaction) {
       datastore.runInTransaction(callable2, readOnlyOptions);
       fail("Expecting a failure");
     } catch (DatastoreException expected) {
-      assertEquals(3, ((DatastoreException) expected.getCause()).getCode());
```
Comment (danieljbruce) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:1623]: It looks like `assertDatastoreException` checks for `expected.getCode()`, but this statement checks for `((DatastoreException) expected.getCause()).getCode()`. If these two methods return different things then should we leave this assert statement here? If these two methods are the same thing then why does the expected code change from 3 to 0?

**Review by danieljbruce — COMMENTED** [review]
Just curious about one thing here.

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java (hunk 1 above)
Comment (kolea2) [reply google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java:1623]: GREAT catch! I've fixed this, PTAL

**Review by danieljbruce — APPROVED** [review]

**Pull request merged**